### DISCUSSION TITLE
New Linux desktop (?)

### DISCUSSION THREAD

**Discussion by dmitrykm** [AvaloniaUI/Avalonia#8292] [Ideas]
I admire the great work you're doing guys. This issue is a long term idea proposal. 

A framework like Avalonia is a big, big, big undertaking. It already demands full-time effort. 

Best tools and frameworks are usually created in the context of and out of the need to solve some bigger applied problems. One example is the C programming language that was created because its author[s] needed to develop operating systems. 

As a big framework, Avalonia needs a powerful application point and a steady "fulcrum" to use as a reference point. One such candidate is a new Linux desktop. Existing Linux desktops are all outdated and UI is one of the weakest points of modern Linux distributions. 

Years ago, one of the `theverge` users, Sputnik8, designed a Windows Desktop UI Concept (https://www.theverge.com/2012/2/24/2822891/windows-desktop-ui-concept). Just one of the images:
![windows_desktop_ui_concept_by_kgbstyle-d4rn113](https://cloud.githubusercontent.com/assets/1682890/26397275/72a1ac38-407e-11e7-81bf-57affd43c7f9.jpg)

What Sputnik8 did back then, is still very relevant. Microsoft itself failed to fully embrace and utilize the potential of its Metro style in its desktop UI, due to overfocusing on "mobile-first" experience. So there is an opportunity here to fix what everyone else has failed to do. 

The **Desktop Of The Future** will be written entirely atop a vector UI framework. It will fully support "responsive layout" and different screen sizes, and will allow easy styling, customization, and scaling. Old school programs and desktops like Windows Explorer are written in C/C++ and are very hard to restyle, scale, and adapt to new requirements. 

If you will be interested, contact a few companies who work with Linux:  SUSE, Canonical, Red Hat, Google. Ask whether they are interested in such a project, big companies usually sponsor useful open source projects they can benefit from. In addition, turn to the Linux community to crowdfund these two projects. This is one of the most obvious ways to move the project forward and invest more effort in it. 

Best wishes.

**Comment by denismaster** [comment]
This is quite fantastic, but it is also important. 
There're already two big Linux toolkits for developing ui, GTK and Qt, and a lot of desktop environments, such as KDE, Ghone Shell, XFCE, MATE, Cinnamon, etc. There is not only one vector for GUI development. Even apps can look terrible if it's written on one toolkit and runned in the desktop based on another toolkit.
There're also a lot of holywars for selecting the best toolkit/desktop. And with that there is no future of deskop linux, in my opinion. Windows and Mac OS X don't have this desktop hell, so they have a big piece of market and consumers. And what about linux? 1 percent. That's a problem. A problem with fragmentation.
So unified, great desktop or even base for this desktop could solve this problem.
